dashboard/test/lib/datastore_cache_test.rb

```diff
@@ -0,0 +1,47 @@
+require 'test_helper'
+require 'dynamic_config/datastore_cache'
+require 'dynamic_config/adapters/memory_adapter.rb'
+
+class DatastoreCacheTest < ActiveSupport::TestCase
+  def setup
+    @data_adapter = MemoryAdapter.new
+  end
+
+  test 'simple get and set' do
+    cache = DatastoreCache.new @data_adapter
+    key = "key"
+    value = [1, 3, 2]
+
+    cache.set(key, value)
+    assert_equal value, cache.get(key)
+  end
+
+  test 'cache is refreshed after expiration time' do
+    key = "key"
+    value = "1, 2, 3"
+
+    @data_adapter.set(key, 'not the real value')
+    cache = DatastoreCache.new @data_adapter, cache_expiration: 0.01
+
+    @data_adapter.set(key, value)
+    sleep 0.1
+
+    assert_equal value, cache.get(key)
+  end
+
```

Comment:
Could you add a test case that the dynamic config recovers when the store is transiently unavailable during startup?